app/src/main/java/com/davidtomas/taskyapp/features/agenda/data/logout/LogoutRepositoryImpl.kt

```diff
@@ -0,0 +1,19 @@
+package com.davidtomas.taskyapp.features.agenda.data.logout
+
+import com.davidtomas.taskyapp.core.domain._util.Result
+import com.davidtomas.taskyapp.core.domain.model.DataError
+import com.davidtomas.taskyapp.features.agenda.data.logout.remote.api.LogoutService
+import com.davidtomas.taskyapp.features.agenda.domain.repository.LogoutRepository
+import com.davidtomas.taskyapp.features.auth.data.local.TaskyDataStore
+
+class LogoutRepositoryImpl(
+    private val logoutService: LogoutService,
+    private val taskyDataStore: TaskyDataStore
+) : LogoutRepository {
+
+    override suspend fun logout(): Result<Unit, DataError.Network> {
+        return logoutService.logout().also {
+            taskyDataStore.deleteToken()
+        }
```

Comment:
What if the coroutine is cancelled after making the logout call and before deleting the token? Then you'll have an invalid state. I'd make sure to execute the deleteToken call in an application wide scope you can inject here